test/functional/test_framework/test_framework.py

```diff
@@ -627,19 +625,32 @@ def connect_nodes(self, a, b, *, peer_advertises_v2=None, wait_for_connect: bool
         if not wait_for_connect:
             return
 
+        # Use subversion as peer id
+        from_connection_subver = from_connection.getnetworkinfo()['subversion']
+        to_connection_subver = to_connection.getnetworkinfo()['subversion']
+
+        def find_conn(node, peer_subversion, inbound):
+            return next(filter(lambda peer: peer['subver'] == peer_subversion and peer['inbound'] == inbound, node.getpeerinfo()), None)
+
         # poll until version handshake complete to avoid race conditions
         # with transaction relaying
```

Comment:
This comment now doesn't really make sense to me?

Maybe
```suggestion
        # wait until peers have completed a version handshake and so match the
        # expected connection subversion
```

Reply:
> This comment now doesn't really make sense to me?

Why? the procedure behavior is the same.

Reply:
I think just the way it read with the old impl vs the new, "waiting until we find a peer connection matching some predicate" rather than "wait until the peer has a non-zero version" . But on second thoughts you're right it's not functionally different so disregard my suggestion

Reply:
Ok. The "version handshake" wording refers to the complete initial negotiation round.
(1) outbound send version, (2) inbound send version, (3) inbound send verack, (4) outbound send verack.